pyvista/utilities/misc.py

```diff
@@ -151,3 +152,14 @@ def can_create_mpl_figure():  # pragma: no cover
 
 
 vtk_version_info = VTKVersionInfo()
+
+
+def set_pickle_format(format: str):
+    """Set the format used to serialize :class:`pyvista.DataObject` when pickled."""
+    supported = {'xml', 'legacy'}
+    format = format.lower()
```

Comment:
Regarding the API:

1. Do we prefer that users use `set_pickle_format()`? Or can they also change  `pyvista.PICKLE_FORMAT` directly?
2. We check stuff in `set_pickle_format()`, but then if this is preferred, shouldn't we consider `pyvista.PICKLE_FORMAT` to be private, and also provide a getter?
3. Can we document somewhere that there are these two format options, and the preferred way of choosing? Especially since we default to legacy, it would be nice to make the opt-in newer feature discoverable. An example perhaps?

Reply:
On 1. and 2. - The user can set `pyvista.PICKLE_FORMAT` directly, or by using the `set_pickle_format()` method. If it is set manually to an invalid value, an error will be thrown pickling. This error at pickle-time is probably sufficient on its own, so we can probably remove the `set_pickle_format()` method.

On 3. - Yes, I will add a note in the docs on how this works. Clearly it is confusing, because as written, it's currently not opt-in to use the new XML readers/writers :) - the XML readers/writers are used by default. Perhaps you were confused by the comment in the `__setstate__` method; this comment states that a pickled is object is assumed to be pickled in a "legacy" format only if its dict does not contain a `PICKLE_FORMAT` key. This is for backwards compatibility, as old pickled objects will not have this key.

Reply:
That's correct, I didn't pay enough attention, sorry 😅 your design is perfect.